### DISCUSSION TITLE
TIVO custom remote....

### DISCUSSION THREAD

**Discussion by pfuller88** [GerryDazoo/Slinger#294] [General]
I started a new thread for this as the other one got long and mixed with SKY.   Here are the changes to the code for my TIVO htlm custom remote:

-Added a readme.txt file
-Cleaned up coding
-Changed dropdown style (if you installed the prior version you can delete the tivoremotefiles folder on your server and download the new one in this package (fewer files).
-renamed main html to TIVOremotePF.html to avoid conflict with base tivo remote - see readme for installing.

Feel free to edit and customize as you wish. 

[TIVOremotePF.zip](https://github.com/GerryDazoo/Slinger/files/10005845/TIVOremotePF.zip)

**Comment by GerryDazoo** [comment]
I'll stick this in the CudtomRemotes folder

**Comment by blasco403** [comment]
Hello,

I got the remote to work with a config.ini file for a single slingbox. The html file and the directory for the tivoremotefiles is in the root of the C:\Singer-main directory where I run the windows version of slinger from.

I am using the latest version you posted in this discussion.

I am trying to get this to work with a unified config as I have a SB 350 and 500. Both of them are connected to the same type of STB so my wife and I can watch different TV at our place in Mexico - therefore the remotes are also identical.

My STB is different (Shaw version of Comcast) but I will edit the html to send correct codes once I can get it working.

Any assistance you can provide would be greatly appreciated.

Thanks

Ron

***************************************************************************

I was testing with my SB 500 as per this config.ini:

[SLINGBOXES]
sb1=350
sb2=500

[350]
sbtype="350/500/M1"
password=xxxxxxxx
ipaddress=192.168.0.4
port=5211
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=55
IframeRate=5
VideoSource=1
maxstreams=2
;StartChannel=155
;RemoteLock=yes
Remote=remote350.txt

[500]
sbtype="350/500/M1"
password=xxxxxxxxxx
ipaddress=192.168.0.221
port=5201
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=55
IframeRate=5
VideoSource=2
maxstreams=2
;StartChannel=155
;RemoteLock=yes
;Remote=remote500.txt
Remote=TIVORemotePF.html

[SERVER]
port=58048
maxremotestreams=2
URLbase=shaw
enableremote=yes

This is the output I get from the slinger application:

The first connection to a remote is for my 350 (old standard txt file based) which works fine.
The second connection to a remote is to my 500 that has the TIVO html remote configured. I am using the latest version you posted in this discussion. As can be seen below, it errors when trying to load the files in the tivoremotefiles directory. The style sheet, png and jpg files cannot be found.

Version : 3.08e Running on Windows-10-10.0.19045-SP0 pid= 10192
Using config file config.ini
Connection Manager Running on port 58048 with 2 max streams using URL shaw.
BOXES [('sb1', '350'), ('sb2', '500')]
Streamer Running:  2 config.ini 500 500 58048 8388608
Streamer Running:  2 config.ini 350 350 58048 8388608
starting up on  port 58048 8388608
Checking for slingbox at ('192.168.0.221', 5201)
Checking for slingbox at ('192.168.0.4', 5211)
('192.168.0.4', 5211) OK
350 Using slingbox at  ('192.168.0.4', 5211)
Streamer:  350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
('192.168.0.221', 5201) OK
500 Using slingbox at  ('192.168.0.221', 5201)
Streamer:  500 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for 350
Reading Custom Remote definition from remote350.txt
Building page for 500
Reading Custom Remote definition from TIVORemotePF.html
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
11/14/2022, 14:43:38.791   RemoteControl connection from ('192.168.0.10', 59278)

Starting remote control stream handler for  ('192.168.0.10', 59278) to port 58049
Remote Control Connected
GET Remote/350
Exiting Remote Control Stream Handler for ('192.168.0.10', 59278)
11/14/2022, 14:43:47.151   RemoteControl connection from ('192.168.0.10', 59284)

Starting remote control stream handler for  ('192.168.0.10', 59284) to port 58049
Remote Control Connected
GET Remote/500
Exiting Remote Control Stream Handler for ('192.168.0.10', 59284)
11/14/2022, 14:43:47.463   RemoteControl connection from ('192.168.0.10', 59285)

Starting remote control stream handler for  ('192.168.0.10', 59285) to port 58049
11/14/2022, 14:43:47.47   RemoteControl connection from ('192.168.0.10', 59286)
Remote Control Connected
GET Remote/tivoremotefiles/layout.css

Starting remote control stream handler for Error: Remote URL needs to be one of:
 ('192.168.0.10', 59286) to port 58049
/Remote/350
Remote Control Connected
/Remote/500
GET Remote/tivoremotefiles/tivoimage.jpg
Exiting Remote Control Stream Handler for ('192.168.0.10', 59285)
Error: Remote URL needs to be one of:
/Remote/350
/Remote/500
Exiting Remote Control Stream Handler for ('192.168.0.10', 59286)
11/14/2022, 14:43:47.760   RemoteControl connection from ('192.168.0.10', 59287)

Starting remote control stream handler for  ('192.168.0.10', 59287) to port 58049
Remote Control Connected
GET Remote/tivoremotefiles/tvpower.png
11/14/2022, 14:43:47.77   RemoteControl connection from ('192.168.0.10', 59288)
Error: Remote URL needs to be one of:
/Remote/350

Starting remote control stream handler for /Remote/500
 ('192.168.0.10', 59288) to port 58049
Exiting Remote Control Stream Handler for ('192.168.0.10', 59287)
Remote Control Connected
GET Remote/tivoremotefiles/tivobutton.png
Error: Remote URL needs to be one of:
/Remote/350
/Remote/500
Exiting Remote Control Stream Handler for ('192.168.0.10', 59288)

**Reply by GerryDazoo** [reply]
You have an old html file.   
In the html you'll  see     ./file_name  in the request. Remove the leading ".".

**Reply by robosmith** [reply]
@blasco403 WRT your Remote HTML for Tivo; "SkipF" button skips forward twice. and ">>2m" button skips forward 4 times. None of the other buttons execute 2x; how can I change that For SkipF? I cannot find where 2x execute is in the code. Thanks

**Reply by robosmith** [reply]
@blasco403 it looks like the 308e code release may have fixed the double execute of SkipF

**Reply by blasco403** [reply]
Thanks, I am using the 3.08e code and all is well, I have edited the html and images for my specific STB from Shaw Canada (comcast/xfinity).

It is working great.

I am using a config with 2 SB's and a different image for my two remotes so I can tell which SB the remote is for.

Two .html files and 2 sub folders.

I will bundle up all the files and post in case others have Shaw as their provider.

**Comment by pfuller88** [comment]
Hi Gerry,

Interestingly, I also tried to setup a unified config for my one slingbox to have different resolutions.   It works but the TIVOremotePF does not work, it connects to the remote but no formatting?  I switch back to the single ini and the remote works fine.  screenshot below as well as my unified code...

[SLINGBOXES]
sb1=500high
sb2=500med
sb3=500low
[500high]
sbtype="350/500/M1/M2"
password=###########
ipaddress=192.168.0.225
port=5201
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
Remote=TIVORemotePF.html
[500med]
sbtype="350/500/M1/M2"
password=###########
ipaddress=192.168.0.225
port=5201
Resolution=5
FrameRate=30
VideoBandwidth=1000
VideoSmoothness=63
VideoSource=1
Remote=TIVORemotePF.html
[500low]
sbtype="350/500/M1/M2"
password=###########
ipaddress=192.168.0.225
port=5201
Resolution=1
FrameRate=30
VideoBandwidth=500
VideoSmoothness=63
VideoSource=1
Remote=TIVORemotePF.html
[SERVER]
port=8080
maxstreams=2
enableremote=yes
URLbase=condo

![image](https://user-images.githubusercontent.com/117112102/202473434-f62c71b9-9cd7-4322-aae7-4f212a3cfd85.png)

**Reply by GerryDazoo** [reply]
It's not loading the JPEG for the remote. Post your logs. You should see a message from the server saying where it is looking for it. Did you unzip all the files into the folder where the server is running? It should have generated a subfolder called something like "tivoremotefiles"

**Comment by pfuller88** [comment]
I have all the files correctly placed, as I mentioned, when I switch the config.ini back to the non-unified ini,  pointing at the same TIVOremotePF.html file, it works fine.   Silly question, where is the log file on linux?

**Reply by blasco403** [reply]
Hey Gerry,

I think I am using the latest posted version. I tried the one from this thread as well as the one posted in customremotes and get the same result as pfuller88. It works with a single SB configured in config.ini but does not format properly when using a "unified" config.ini.

Errors I receive are noted in my earlier post.

Thanks.

**Reply by GerryDazoo** [reply]
Did you make the code change in the HTML, This was an issue with the unifed config until the '." was removed from in front of all the ./filename links in the html file. Maybe the fixed version never found it's way into github.

**Reply by blasco403** [reply]
I have not had time to test yet, will do so later today.

Which version do you suggest I edit - the one posted in this thread or the one in the customremotes on GitHub?

Thanks.

**Reply by GerryDazoo** [reply]
I think they are the same.

**Comment by pfuller88** [comment]
Here is my browser error message, changed the URL to hide.  Seems like it is looking in the right location?

Request URL: http://www.pbsdfdssdfsdf.net:8080/Remote/tivoremotefiles/tivobutton.png
Request Method: GET
Status Code: 404 NOT FOUND
Remote Address: -------:8180
Referrer Policy: strict-origin-when-cross-origin
Connection: close
Content-Length: 207
Content-Type: text/html; charset=utf-8
Date: Thu, 17 Nov 2022 14:57:01 GMT
Server: Werkzeug/2.2.2 Python/3.10.6
Accept: image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9
Connection: keep-alive
Cookie: LastBootTime=1637265342
Host: www.psdfsdfsdfdsf.net:8080
Referer: http://www.pb.sdfsffsdf.net:8080/Remote/500high
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 Edg/107.0.1418.42

**Reply by GerryDazoo** [reply]
Yes, the "Remote" shouldn't be in the URL requested by your browser. Please make the change requested in the html file.

**Comment by pfuller88** [comment]
I think we're getting closer to solving this.  I do not have the word "Remote" in the path in the HTML file.   The http I use to get to the remote is  below.  Not sure where the "remote" directory is being added?

http://www.pbsdfsdfsdfds.net:8080/Remote/500high

**Reply by GerryDazoo** [reply]
It doesn't need to be solved. I told you what was wrong and how to fix it.
It's being added by your browser because the HTML is incorrect. It's that pesky '.' that's causing your browser to add the "Remote"

**Comment by pfuller88** [comment]
Please note,  as per the other users comments above, the TIVOremote.html works fine when not using the unified ini setup.

**Reply by GerryDazoo** [reply]
It's not server code. It's the HTML.

**Comment by pfuller88** [comment]
here is a sample of what the HTML has for an image...

img class="powerimage" src="tivoremotefiles/tvpower.png" alt="power"

**Reply by GerryDazoo** [reply]
Dang. Not what I was expecting. He took out the . and the /
Please change src="  to src="/tivoremotefiles/tvpower.png" alt="power"

Sorry about that.

**Reply by blasco403** [reply]
Thanks Gerry.

The html did not have "./" as you just eluded to.

I was testing as you were posting with just adding a "/".

The remote now works fine with a unified config.

- add a "/" in front of the "tivoremotefiles" in the HTML.

**Comment by pfuller88** [comment]
Yes, thanks, the "/tiveremotefiles/..." works for the unified ini.

**Comment by blasco403** [comment]
I updated the ZIP file posted above with the edit to the HTML.

This version now works with a unified_config setup.

[TIVOremotePF.zip](https://github.com/GerryDazoo/Slinger/files/10033075/TIVOremotePF.zip)

**Reply by GerryDazoo** [reply]
THX. I'll post it in the repository

**Reply by RandomMan719** [reply]
Gerry, I thought we had that fixed last week.  Thank you for adding it to the repository.  I will download and try it.

**Reply by GerryDazoo** [reply]
Me too. Maybe I missed a file upload in that thread.

**Comment by blasco403** [comment]
Hello,

I have edited the HTML file to map my STB remote codes properly (Shaw Canada - Xfinity Comcast) and all is working good except when I use the Select Channel dropdown list. My STB requires you hit OK after you have selected the channel you want. When I pick one of the channels from the dropdown, it sends the correct keys but does not change to the channel because it expects an "OK" as well.

I am not a programmer, but can usually edit things to get them going, I tried a few things as I see it uses the SendDigits2 and Send Digits3 functions depending on if you are using the full or mini remote. I cannot seem to make it send the digits and then on "OK".

Any guidance would be greatly appreciated.

Thanks.

**Reply by GerryDazoo** [reply]
Just did a quick peek at the code. I think you can just add a
sendKeys('42','sel')
To the end of SendDIgits3 function.

**Reply by GerryDazoo** [reply]
Or, you can change the channel numbers to add the OK. Use keycodes instead of channel digits like so:
9,18,9,42
will tune to 101 and send OK. i.e.

{"id":"909","text":"Starz Edge"}  -> {"id":"17,18,17,42","text":"Starz Edge"}

digits2keycodes = [18,9,10,11,12,13,14,15,16,17]

**Reply by pfuller88** [reply]
note, if you change sendDigits(3) as per above, by adding sendKeys('42','sel'),  change sendDigits(2) function as well.  I have not tried this solution yet.   sendDigits(2) is for the dropdown on the main remote and sendDigits(3) is for the dropdown on the mini remote.

**Reply by blasco403** [reply]
Hello,

I got it to work by adding the sendKeys example Gerry suggested. I added to both sendDigits2 and 3 functions:

		function sendDigits2(){
			var digit = $('#Digit2').val();
				//console.log("Sending: " + "Digits" + "(" + digit + ")");
			$.ajax({
				data: {Digits : digit},
				type: "POST",
				cache: false,
				success: function (response) {
				// console.log(response);
				}
			});
		sendKeys('42','sel')
		}
		function sendDigits3(){
			var digit = $('#Digit3').val();
			$.ajax({
				data: {Digits : digit},
				type: "POST",
				cache: false,
				success: function (response) {
				}
			});
		sendKeys('42','sel')
		}

Thanks for your assistance.

**Comment by Lope0407** [comment]
Hello, I use a TiVo Bolt with my Slingbox 500.  Its the only TiVo model I've ever had, so I don't know if this is strictly a Bolt issue or if it applies to other models.  On the Bolt, when you want to reboot the TiVo via the remote control (in the Help menu) after clicking Restart it asks you to verify by pressing Thumbs-Down,Thumbs-Down,Thumbs-Down,Enter.  The existing TiVo remotes people have built are great, but without the Thumbs-Down button, someone could get stuck away from home with no way to reboot their TiVo.  If you are still tinkering with the TiVo remotes, would you mind adding that key (code=79)?  Thanks.

**Reply by tom76017** [reply]
As an alternate for TiVos, take a look at https://github.com/GerryDazoo/Slinger/discussions/30 and https://github.com/GerryDazoo/Slinger/discussions/84.   They're not directly connected to Slinger, but I use my phone or laptop to control my Slinger session.

**Reply by pfuller88** [reply]
I would think there are key code numbers for those buttons, you could replace a few of the favourite channels with them in the html code.  As an aside, I have my remote sling box plugged into a cheap smart plug which I can turn on and off from anywhere.

**Comment by cupanther** [comment]
I have a TiVo Series 3 and to enter a digital antenna channel you press the advance button for the dash to get the correct digital channel.  I have not been able to accomplish this with a slingbox 350.  Am I not using the correct code for the "dash" or does the slingbox 350 not support this command?

**Reply by pfuller88** [reply]
You can try finding the correct keycode and add this to a button on the remote html or remote text file (or replace one of the other buttons.   I found this link to the codes available for the S3, trial and error, not sure which one is for "advance" or if it will work.

[http://[autoupdate.sling.com/remotedb/Remote-keymaps/TiVoS3.idmap.tivo.xml](http://autoupdate.sling.com/remotedb/Remote-keymaps/TiVoS3.idmap.tivo.xml)](url)

**Reply by pfuller88** [reply]
just notice in the keycode link above, code 51 says "dash", good starting point?

**Reply by cupanther** [reply]
I found it in the map.  It is code 30 for "Quick Skip".

Thanks for your help.

**Comment by Drano32** [comment]
Hi all - I'm a Mac user, and successfully streaming on all my Slingboxes using the unified_config setup (thanks to TheWeatherGuy's Mac setup guide, which worked great for me).  I'm excited that the steaming is working on all my boxes.  But I'm stuck on getting any remote to work.  

All my inputs are Tivos.  I've put the TIVORemotePF.html file in the same folder and level as my unified_config.ini and slingbox_server.py files.  Same for the tivoremotefiles folder.  Finally, I've defined the Remote=TIVORemotePF.html file each box configuration in the unified config file.  

When I try to access the remote in my browser, I get a connection refused error message.  Logs below.  Any ideas of what I'm doing wrong or what I should change?

**Reply by Drano32** [reply]
[SLINGBOXES]
sb1=Basement500
sb2=Upstairs500
sb3=UpstairsM2

[Basement500]
sbtype="350/500/M1"
password=xxxxx
ipaddress=10.0.1.43
port=5201
Resolution=15
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
Remote=TIVORemotePF.html

[Upstairs500]
sbtype="350/500/M1"
password=xxxxxxxx
ipaddress=10.0.1.18
port=5203
Resolution=15
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
Remote=TIVORemotePF.html

[UpstairsM2]
sbtype="350/500/M1/M2"
password=xxxxxxxx 
ipaddress=10.0.1.34
port=5324
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
Remote=TIVORemotePF.html

[SERVER]
port=8080
maxremotestreams=10
URLbase=slingbox
enableremote=yes

**Reply by Drano32** [reply]
Starting remote control stream handler for  ('10.0.1.240', 58418) to port 8081
Exception in thread Thread-26 (remote_control_stream):
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/xxxxxx/slingbox_server.py", line 995, in remote_control_stream
    remote_control_socket.connect(('127.0.0.1', http_port ))  ## Send Packets to Flask
ConnectionRefusedError: [Errno 61] Connection refused

**Reply by Drano32** [reply]
Slinginfo  "350/500/M1" 15 60 10.0.1.18 5203 3072 3
Upstairs500 Starting Stream for  10.0.1.240:58600
Connecting... ('10.0.1.18', 5203) Control
Upstairs500 Selecting Video Source 1
Keycodes= [4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 26, 27, 28, 29, 30, 31, 32, 35, 36, 37, 38, 39, 40, 41, 42, 46, 50, 72, 73, 74, 75, 76, 77, 78, 79]
VideoParameters: Resolution= 15 FrameRate= 60 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('10.0.1.18', 5203) Stream
Upstairs500 Stream started at 11/20/2022, 18:34:13.388  744 0

**Reply by pfuller88** [reply]
what URL are you using,  should be like. …   http://yourIPaddress/Remote/Basement500

**Reply by Drano32** [reply]
First of all thank you for your work on the Tivo remote!  Yes, that's the exact address except that I am using:

http://myIPaddress:8080/Remote/Basement500

Note that I've also tried a few of the other Tivo remote.txt files (placing them in the same folder as my config and server files) and the same error occurs.

**Reply by GerryDazoo** [reply]
I've never seen this one before.
remote_control_socket.connect(('127.0.0.1', http_port )) ## Send Packets to Flask
ConnectionRefusedError: [Errno 61] Connection refused

I'm just winging it here. Can you try changing your [SERVER] port to something other than 8080 or 8081. I'm thinking some other software might be binding to the port (8081) that my software needs.

**Reply by Drano32** [reply]
Thank you Gerry, for both this response and all your work on this project. 

I also noticed the weird mention of the 127.0.0.1 request. I also thought it was interesting that port 8081 was referenced - certainly 8081 is not a direct part of my configuration (but you may know why an adjacent port somehow entered the picture!). I’ll try a different port than 8080 and report back. Would it also make sense to try a non-unified config file (I’m sure that’s possible on a Mac even if not shown in TheWeatherGuy’s guide and terminal command examples). 

Just so you know the steaming works perfectly on all three boxes, including remotely from outside my network.  Have never been able to get any remote to work yet.

**Reply by GerryDazoo** [reply]
8081 is used internally by my software to send data to the flask module for remote control. The connection request is sent to localhost (127.0.0.1:server_port+1.
Using a non unified config will make you use two different server ports so basically the same as changing the [SERVER] port in the unified config.

**Reply by Drano32** [reply]
Hi Gerry - attempted a few additional, different ports (including 58048 as seen below).  Streaming works fine as usual but still get the same error.  Based on this error message do you think I have the html file and supporting files in the correct place, but something happens?  Could I have missed something very basic here?  

I have the remote files in the same folder as the config and server files.  

Was there something else basic that I've missed?  Using the http://ipaddress:58048/Remote/Basement500 address in my Firefox browser.

**Reply by Drano32** [reply]
Slinginfo  "350/500/M1" 15 60 10.0.1.43 5201 3072 3
Basement500 Starting Stream for  10.0.1.240:57395
Connecting... ('10.0.1.43', 5201) Control
Basement500 Selecting Video Source 1
Keycodes= [24, 27, 28, 33, 38, 39, 40, 41, 42]
VideoParameters: Resolution= 15 FrameRate= 60 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('10.0.1.43', 5201) Stream
Basement500 Stream started at 11/21/2022, 10:40:25.26  744 0
....11/21/2022, 10:41:07.474   RemoteControl connection from ('10.0.1.240', 57414)

Starting remote control stream handler for  ('10.0.1.240', 57414) to port 58049
Exception in thread Thread-5 (remote_control_stream):
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/arthurh/slingbox_server.py", line 995, in remote_control_stream
    remote_control_socket.connect(('127.0.0.1', http_port ))  ## Send Packets to Flask
ConnectionRefusedError: [Errno 61] Connection refused

**Reply by GerryDazoo** [reply]
I know this is going to sound funny but you haven't by any chance not enabled the remote in your config.ini
Please post that and ALL the server output not just the part starting the steam

**Reply by Drano32** [reply]
Gerry, yes, I think I've enabled - see below.  Will also post the complete server output.

**Reply by Drano32** [reply]
[SLINGBOXES]
sb1=Basement500
sb2=Upstairs500
sb3=UpstairsM2

[Basement500]
sbtype="350/500/M1"
password=xxxxxx
ipaddress=10.0.1.43
port=5201
Resolution=15
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
Remote=TIVORemotePF.html

[Upstairs500]
sbtype="350/500/M1"
password=xxxxxx
ipaddress=10.0.1.18
port=5203
Resolution=15
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
Remote=TIVORemotePF.html

[UpstairsM2]
sbtype="350/500/M1/M2"
password=xxxxxxx
ipaddress=10.0.1.34
port=5324
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
Remote=TIVORemotePF.html

[SERVER]
port=58048
maxremotestreams=10
URLbase=slingbox
enableremote=yes

**Reply by Drano32** [reply]
In the complete log (below) I do see a mention that there is no flask module ("ModuleNotFoundError: No module named 'flask').  Could that be it?

**Reply by Drano32** [reply]
% python3 slingbox_server.py unified_config.ini
Version : 3.08e Running on macOS-11.6.8-x86_64-i386-64bit pid= 47507
Using config file unified_config.ini
Connection Manager Running on port 58048 with 10 max streams using URL slingbox.
BOXES [('sb1', 'Basement500'), ('sb2', 'Upstairs500'), ('sb3', 'UpstairsM2')]
Warning TCP buffering might not be sufficent for reliable streaming
Warning TCP buffering might not be sufficent for reliable streaming
Warning TCP buffering might not be sufficent for reliable streaming
Streamer Running:  10 unified_config.ini Basement500 Basement500 58048 7340032
Streamer Running:  10 unified_config.ini Upstairs500 Upstairs500 58048 7340032
Traceback (most recent call last):
Streamer Running:  10 unified_config.ini UpstairsM2 UpstairsM2 58048 7340032
  File "/Users/arthurh/slingbox_server.py", line 1430, in <module>
Warning TCP buffering might not be sufficent for reliable streaming
Checking for slingbox at ('10.0.1.18', 5203)
Checking for slingbox at ('10.0.1.34', 5324)
Checking for slingbox at ('10.0.1.43', 5201)
    from flask import Flask, render_template, request, render_template_string, abort, send_file
ModuleNotFoundError: No module named 'flask'
('10.0.1.18', 5203) OK
starting up on  port 58048 7340032
Upstairs500 Using slingbox at  ('10.0.1.18', 5203)
('10.0.1.34', 5324) OK
Streamer:  Upstairs500 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
UpstairsM2 Using slingbox at  ('10.0.1.34', 5324)
Streamer:  UpstairsM2 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
('10.0.1.43', 5201) OK
Basement500 Using slingbox at  ('10.0.1.43', 5201)
Streamer:  Basement500 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
11/21/2022, 11:17:49.39   Streaming request from ('10.0.1.240', 58693)

Slinginfo  "350/500/M1" 15 60 10.0.1.43 5201 3072 3
Basement500 Starting Stream for  10.0.1.240:58693
Connecting... ('10.0.1.43', 5201) Control
Basement500 Selecting Video Source 1
Keycodes= [24, 27, 28, 33, 38, 39, 40, 41, 42]
VideoParameters: Resolution= 15 FrameRate= 60 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('10.0.1.43', 5201) Stream
Basement500 Stream started at 11/21/2022, 11:17:51.692  744 0
...11/21/2022, 11:18:22.998   RemoteControl connection from ('10.0.1.240', 58706)

Starting remote control stream handler for  ('10.0.1.240', 58706) to port 58049
Exception in thread Thread-5 (remote_control_stream):
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/arthurh/slingbox_server.py", line 995, in remote_control_stream
    remote_control_socket.connect(('127.0.0.1', http_port ))  ## Send Packets to Flask
ConnectionRefusedError: [Errno 61] Connection refused

**Reply by GerryDazoo** [reply]
Yes, most certainly that is the problem. Flask in the module that handles the HTTP remote requests.
You could have saved us alot of time by mentioning that error in the first place. 
And you didn't read the documentation that says you need flask for the remote to work.

**Reply by Drano32** [reply]
Thank you.  I did read the documentation and followed all the Mac installation steps in the guide posted by TheWeatherGuy.  Never noticed the error in the log, sorry.  I will figure out how to install (or re-install) flask.  Greatly appreciate all of your time.

**Reply by GerryDazoo** [reply]
This is from the readme.
# you don't need the flask module if you're not using the embedded web server to server out the Remote Control page. Disable the remote in your config.ini file set enableremote=no.
    sudo pip install flask

PLease alert M. MAC if this is over looked in his documentation.
